### PR TITLE
Recipe improvements

### LINKED ISSUE #778: [Bug] LCT: Joining a world with a LCT with a valid recipe gives the smarter than the teacher advancement.
**Describe the bug**
Joining a world with a LCT with a valid recipe gives the smarter than the teacher advancement.

**To Reproduce**
1. add a valid recipe (like CH4) to a LCT.
2. Leave the world
3. Rejoin the world

**Expected behavior**
LCT should recognise the recipe as a valid recipe, without triggering the advancement.

### PR DESCRIPTION
**Pull request by pixar02**
- [x] fixes https://github.com/ScicraftLearn/Minelabs/issues/263
- [x] fixes https://github.com/ScicraftLearn/Minelabs/issues/522 
- [x] Overal recipe improvements (time to craft, container requirement and craft amount are configurable in json file)
- [x] Fix strange clear button interraction
- [x] WIP fixes https://github.com/ScicraftLearn/Minelabs/issues/767
- [x] fixes #778

### CODE REVIEW

File: src/main/java/be/minelabs/recipe/molecules/Molecule.java [hunk 1]
```diff
@@ -5,13 +5,19 @@
 public class Molecule extends PartialMolecule {
     private final Item item;
 
-    public Molecule(MoleculeGraph structure, Item item) {
+    private final Integer count;
+
+    public Molecule(MoleculeGraph structure, Item item, Integer count) {
         super(structure);
         this.item = item;
+        this.count = count;
```
Comment (JoeyDP) [inline src/main/java/be/minelabs/recipe/molecules/Molecule.java:13]: What is the use case for this count?

Comment (JoeyDP) [reply src/main/java/be/minelabs/recipe/molecules/Molecule.java:13]: Seems like it's only part of the recipes at first sight. Then the recipe might be a better way to store it? I.e. immediately as a result itemstack.

**Review by JoeyDP — CHANGES_REQUESTED** [review]

**Comment by pixar02** [comment]
> I'm not sure #522 was actually still needed or even possible, since it's Screen info that needs to be synced and not really block entity data (stored in nbt). It's been to long and I don't remember why the recipe/ingredients needed to be synced in the first place though. Let's just leave it if this works. Sorry to be too nitpicky about it.

We need it for the rendering the atoms required. (atom symbols) (When a valid recipe is found and the IoInventory slots are unlocked)

**Comment by JoeyDP** [comment]
Hmm and I guess the Client side is not able to compute it independently? I'm wondering how a crafting table does this, because if also needs to show the result item, which is a similar task.

File: src/main/java/be/minelabs/recipe/molecules/Molecule.java (hunk 1 above)
Comment (pixar02) [reply src/main/java/be/minelabs/recipe/molecules/Molecule.java]: If we want more items as an output. 
If an recipe gives 2 items then the count is 2.

**Review by smtdm — APPROVED** [review]
Works like a charm!

File: src/main/java/be/minelabs/recipe/molecules/Molecule.java (hunk 1 above)
Comment (JoeyDP) [reply src/main/java/be/minelabs/recipe/molecules/Molecule.java]: Do you also think it would fit better in the `MoleculeRecipe` then?

**Review by JoeyDP — CHANGES_REQUESTED** [review]

**Review by JoeyDP — APPROVED** [review]

**Pull request merged**